test/cpp/api/tensor.cpp

```diff
@@ -211,6 +260,15 @@ TEST(TensorTest, ContainsCorrectValuesForManyValuesVariable) {
   tensor = torch::tensor({1.5, 2.25, 3.125});
   ASSERT_TRUE(tensor.is_variable());
   ASSERT_EQ(tensor.numel(), 3);
+  ASSERT_EQ(tensor.sizes(), std::vector<int64_t>({3}));
+  ASSERT_EQ(tensor.dtype(), at::kDouble);
```

Comment:
To match the python API, I'd expect both of these to have dtype float32. Is there a reason why we aren't doing this?

Reply:
Yes I haven't got around to add `torch::set_default_dtype` yet. I will do it as a follow-up PR.

Reply:
How hard is it to add torch::set_default_dtype? It looks like it might be more efficient to add it and then merge this PR (so we don't have to change dtypes manually to kFloat in this change and then reverse this change in the future)

Reply:
I found that we already have `torch::set_default_dtype` implemented in https://github.com/pytorch/pytorch/blob/d8c368bd62455f2c9d0bf036640a87281db0face/c10/core/DefaultDtype.h#L10
but to change the default dtype to float32 for the `torch::tensor` constructor would be another BC-breaking change, which I think could happen independent of this PR. (When we change the default dtype in that PR, we will only need to change the expected default dtype for test cases such as `torch::tensor({1.5, 2.25, 3.125})` in `tensor.cpp`, and we don't need to change the functional tests / module tests because we explicitly specify their input's dtype (which is good because it doesn't have dependency on what the default dtype is).)

Reply:
Got it, that sounds reasonable. Could we add a comment somewhere (I'm not sure where) about the inconsistency (torch::tensor by default give a double tensor) that links an issue that says we are going to work on changing torch::tensor to give a float tensor by default?

Reply:
Maybe in `torch::tensor`

Reply:
Thanks! I added it above `inline at::Tensor tensor(detail::TensorDataContainer tensor_data_container, const at::TensorOptions& options = {})` in this PR